salt/engines/junos_syslog.py

```diff
@@ -0,0 +1,181 @@
+'''
+Junos Syslog Engine
+==========================
+
+An engine that listen to syslog message from Junos devices,
+extract event information and generate message on SaltStack bus.
+
+:configuration:
+  Example configuration
+    engines:
+      - junos_syslog:
+          port: 516
+
+For junos_syslog engine to receive events syslog must be set on junos device, to forward events.
+This can be done via following configuration:
+	set system syslog host <ip-of-the-salt-device> any any
+
+Here is a sample syslog event which is received from the junos device:
+	' <30>May 29 05:18:12 bng-ui-vm-92 mspd[1492]: No chassis configuration found '
+
+The source for parsing the syslog messages is taken from:
+	https://gist.github.com/leandrosilva/3651640#file-xlog-py
+'''
+
+__author__ = "Nitin Kumar, Rajvi Dhimar"
+
+from twisted.internet.protocol import DatagramProtocol
```

Comment:
Since salt does not depend on twisted, we need to get these imports behind a try/except block. Please see the docker engine as an example: https://github.com/saltstack/salt/blob/develop/salt/engines/docker_events.py#L17